app/autotranslate/server/msTranslate.js

```diff
@@ -0,0 +1,167 @@
+		};
+	}
+
+	/**
+	 * Returns supported languages for translation by the active service provider.
+	 * Microsoft does not provide an endpoint yet to retrieve the supported languages.
+	 * So each supported languages are explicitly maintained.
+	 * @private implements super abstract method.
+	 * @param {string} target
+	 * @returns {object} code : value pair
+	 */
+	getSupportedLanguages(target) {
+		if (this.autoTranslateEnabled && this.apiKey) {
+			if (this.supportedLanguages[target]) {
+				return this.supportedLanguages[target];
+			}
+			const languages = HTTP.get(this.apiGetLanguages);
+			this.supportedLanguages[target] = Object.keys(languages.data.translation).map((language) => ({
+				language,
+				name: languages.data.translation[language].name,
+			}));
+			return this.supportedLanguages[target || 'en'];
+		}
+	}
+
+	/**
+	 * Re-use method for REST API consumption of MS translate.
+	 * @private
+	 * @param {object} message
+	 * @param {object} targetLanguages
+	 * @throws Communication Errors
+	 * @returns {object} translations: Translated messages for each language
+	 */
+	_translate(data, targetLanguages) {
+		let translations = {};
+		const supportedLanguages = this.getSupportedLanguages('en');
+		targetLanguages = targetLanguages.map((language) => {
+			if (language.indexOf('-') !== -1 && !_.findWhere(supportedLanguages, { language })) {
+				language = language.substr(0, 2);
+			}
+			return language;
+		});
+		const url = `${ this.apiEndPointUrl }&to=${ targetLanguages.join('&to=') }`;
+		const result = HTTP.post(url, {
+			headers: {
+				'Ocp-Apim-Subscription-Key': this.apiKey,
+				'Content-Type': 'application/json; charset=UTF-8',
+			},
+			data,
+		});
+
+		if (result.statusCode === 200 && result.data && result.data.length > 0) {
+			// store translation only when the source and target language are different.
+			translations = Object.assign({}, ...targetLanguages.map((language) =>
+				({
+					[language]: result.data.map((line) => line.translations.find((translation) => translation.to === language).text).join('\n'),
+				}),
+			));
+		}
+
+		return translations;
+	}
+
+	/**
+	 * Returns translated message for each target language.
+	 * @private
+	 * @param {object} message
+	 * @param {object} targetLanguages
+	 * @returns {object} translations: Translated messages for each language
+	 */
+	_translateMessage(message, targetLanguages) {
+		let translations = {};
+
+		// There are multi-sentence-messages where multiple sentences come from different languages
+		// This is a problem for translation services since the language detection fails.
+		// Thus, we'll split the message in sentences, get them translated, and join them again after translation
+		const msgs = message.msg.split('\n').map((msg) => ({ Text: msg }));
+		try {
+			translations = this._translate(msgs, targetLanguages);
+		} catch (e) {
+			logger.microsoft.error('Error translating message', e);
+		}
+		return translations;
```

Comment:
```suggestion
		// There are multi-sentence-messages where multiple sentences come from different languages
		// This is a problem for translation services since the language detection fails.
		// Thus, we'll split the message in sentences, get them translated, and join them again after translation
		const msgs = message.msg.split('\n').map((msg) => ({ Text: msg }));
		try {
			return this._translate(msgs, targetLanguages);
		} catch (e) {
			logger.microsoft.error('Error translating message', e);
		}
		return {};
```